lib/injjject.rb

```diff
@@ -0,0 +1,13 @@
+
+class Array
+  def injjject(argument = nil)
+    if argument
```

Comment:
is there a way you could do this without this if and stop the duplications. if you did something like  `starter = (argument ? argument : self.shift )` at the beginning you sould then just self.each